### PR TITLE
Feat/Complete the sfml class

### LINKED ISSUE #21: Complete the SFML class
# :white_check_mark: What's needed
You must complete the SFML class.

## 📋 Subtasks
When the SFML is fully completed:
- [x] Handle events
- [x] Displays entities
- [x] Displays Texts
- [x] Plays sound

## :file_folder: Files required
src/Graphic/SFML/SFML.cpp
src/Graphic/SFML/SFML.hpp

### PR DESCRIPTION
**Pull request by Thomaltarix**
I completed the SFML class with the missions functions:
- Entities Display
- Texts Display
- Music Playing

I Also added Keymaps for the Event inputs.
The reference is in IGraphic.hpp.
For the SFML lib, I defined a keymap tab to visually see the matches.

Too see more details: #21

### CODE REVIEW

File: src/Graphical/SFML/SFML.cpp [hunk 1]
```diff
@@ -7,9 +7,9 @@
 
 #include "SFML.hpp"
 
-SFML_lib::SFML_lib()
+SFMLlib::SFMLlib()
 {
-    if (isWindowOpen() == true)
+    if (isWindowOpen())
```
Comment (mathieurobert1) [inline src/Graphical/SFML/SFML.cpp]: Why did you do that ?

File: src/Graphical/SFML/SFML.hpp [hunk 2]
```diff
@@ -32,4 +34,62 @@ class SFML_lib : public AGraphic {
...
+        std::unordered_map<std::string, sf::Texture> _textures;
+        std::unordered_map<std::string, sf::Font> _fonts;
+        std::unordered_map<std::string, sf::SoundBuffer> _soundBuffers;
+        std::unordered_map<std::string, sf::Sound> _sounds;
+
+        sf::Sprite createSprite(std::shared_ptr<IEntity> entity);
+        sf::Text createText(std::shared_ptr<IText> text);
+        void playSound(std::shared_ptr<ISound> sound, bool loop);
+
+        const std::unordered_map<sf::Keyboard::Key, int> _keyMap = {
```
Comment (mathieurobert1) [inline src/Graphical/SFML/SFML.hpp:46]: Ok, I think it's the best way to do that :+1:

**Review by mathieurobert1 — APPROVED** [review]
It seems good for me :+1:

**Review by theophile-jr — APPROVED** [review]

File: src/Graphical/SFML/SFML.hpp (hunk 2 above)
Comment (Thomaltarix) [reply src/Graphical/SFML/SFML.hpp:46]: Hugo told me that it was a good thing during the bootstrap :+1:

File: src/Graphical/SFML/SFML.cpp (hunk 1 above)
Comment (Thomaltarix) [reply src/Graphical/SFML/SFML.cpp]: It seems better for me, but if you prefer keeping the `== true` i can keep it, just tell me

Comment (mathieurobert1) [reply src/Graphical/SFML/SFML.cpp]: No it's just : why did you check if the window is open at the beginning of the constructor ?
The window is not created yet.

Comment (Thomaltarix) [reply src/Graphical/SFML/SFML.cpp]: In fact, you're right !

Comment (Thomaltarix) [reply src/Graphical/SFML/SFML.cpp]: That's fixed

**Pull request merged**